### PR TITLE
caasp4: rename "storage" role to "nfs" and drop it from default 4-node deployment

### PR DESCRIPTION
**Pull request by colder-is-better**
Fixes: https://github.com/SUSE/sesdev/issues/308

Signed-off-by: Christos Varelas <subzraw@colder.xyz>

### CODE REVIEW

File: seslib/templates/caasp/master.sh.j2 [hunk 1]
```diff
@@ -78,8 +78,10 @@ kubectl create serviceaccount --namespace kube-system tiller
 kubectl create clusterrolebinding tiller-cluster-rule --clusterrole=cluster-admin --serviceaccount=kube-system:tiller
 helm init --service-account=tiller --wait
 
+{% if node_manager.get_by_role('storage') | length >0 %}
```
Comment (smithfarm) [inline seslib/templates/caasp/master.sh.j2:81]: {% if storage_nodes > 0 %}

would be shorter and IMO more elegant

**Comment by smithfarm** [comment]
I don't like that "storage" is used with two different meanings. (For a ceph cluster, it means "OSDs", while for a caasp4 cluster it means "NFS Server".)

We already have an "nfs" role. Would it make sense to change the caasp4 semantics to use "nfs" instead of "storage"?

File: README.md [hunk 2]
```diff
@@ -348,6 +348,14 @@ The DeepSea version shipped with SES5 always deploys Prometheus and Grafana
 instances on the master node, but does not recognize "prometheus"/"grafana"
 roles in policy.cfg.
 
+[3] The `storage` role may also be used when deploying a CaaSP cluster. In that
```
Comment (smithfarm) [inline README.md:351]: change "storage" here to "nfs" (If you agree with my proposal, that is)

File: README.md [hunk 3]
```diff
@@ -348,6 +348,14 @@ The DeepSea version shipped with SES5 always deploys Prometheus and Grafana
...
+case we get a node acting as an NFS server as well as a pod running in the k8s
+cluster and acting as an NFS client, providing a persistent store for other
+(containerized) applications. Please note, though, that we do not need the
+`storage` role when we plan to deploy Rook/Ceph over CaaSP. By default, Rook
+creates OSD pods which take over any spare block devices in worker nodes,
+i.e., all block devices but the first (OS disk) of any given worker.
```
Comment (smithfarm) [inline README.md:357]: The sentence starting with "Please note" would become a separate note.

File: seslib/templates/caasp/master.sh.j2 (hunk 1 above)
Comment (colder-is-better) [reply seslib/templates/caasp/master.sh.j2]: Accepted proposal and introduced some additional changes required from the new usage of role `nfs`.

File: README.md (hunk 2 above)
Comment (colder-is-better) [reply README.md]: Done!

File: README.md (hunk 3 above)
Comment (colder-is-better) [reply README.md]: Done!

File: README.md [hunk 4]
```diff
@@ -329,21 +329,31 @@ The following roles can be assigned:
...
 [1] CAVEAT: sesdev applies the "admin" role to all nodes, regardless of whether
 or not the user specified it explicitly on the command line or in `config.yaml`.
 
-[2] CAVEAT: Do not specify "prometheus"/"grafana" roles for ses5 deployments.
+[2] The `nfs` role may also be used when deploying a CaaSP cluster. In that
+case we get a node acting as an NFS server as well as a pod running in the k8s
+cluster and acting as an NFS client, providing a persistent store for other
+(containerized) applications.
+
+[3] Please note that we do not need the `nfs` role when we plan to deploy
```
Comment (smithfarm) [inline README.md:351]: please move this note to the "storage" role and make this change:

```
+ Please note that we do not need the `nfs` role when we plan to deploy
+ Please note that we do not need the `storage` role when we plan to deploy
```

**Comment by smithfarm** [comment]
@capri1989 This looks good to me and passes my admittedly very basic test, `contrib/standalone.sh --caasp`.

Did you want to review it before I merge it?

**Pull request merged**